### PR TITLE
Add permissions middleware

### PR DESCRIPTION
**Pull request by mikemrm**
Produces an echo middleware which adds a checker function into the
request context that is later used to check if the current actor has
access to the requesting resource and action.

### CODE REVIEW

File: pkg/permissions/permissions.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+		return nil
+	}
+}
+
+// New creates a new Permissions instance
+func New(config Config, options ...Option) (*Permissions, error) {
+	p := &Permissions{
+		enabled:        config.URL != "",
+		client:         http.DefaultClient,
```
Comment (fishnix) [inline pkg/permissions/permissions.go:140]: let's use a sane client with a really short timeout instead of the default client

**Review by fishnix — CHANGES_REQUESTED** [review]
just the one change - otherwise looks good!

File: pkg/permissions/permissions.go (hunk 1 above)
Comment (fishnix) [reply pkg/permissions/permissions.go:140]: (unless you think its better for implementers to pass a client - but that feels like a 🦶 🔫

Comment (mikemrm) [reply pkg/permissions/permissions.go:140]: yeah fair point

**Review by fishnix — APPROVED** [review]

**Pull request merged**